src/zcbor_decode.c

```diff
@@ -133,24 +133,35 @@ static bool value_extract(zcbor_state_t *state,
 	ZCBOR_ERR_IF((state->elem_count == 0), ZCBOR_ERR_LOW_ELEM_COUNT);
 
 	uint8_t additional = ZCBOR_ADDITIONAL(*state->payload);
-	size_t len = additional_len(additional);
-	uint8_t *result_offs = (uint8_t *)result + ZCBOR_ECPY_OFFS(result_len, MAX(1, len));
+	size_t len = 0;
 
-	ZCBOR_ERR_IF(additional > ZCBOR_VALUE_IS_8_BYTES, ZCBOR_ERR_ADDITIONAL_INVAL);
-	ZCBOR_ERR_IF(len > result_len, ZCBOR_ERR_INT_SIZE);
-	ZCBOR_ERR_IF((state->payload + len + 1) > state->payload_end,
-		ZCBOR_ERR_NO_PAYLOAD);
+	(void)indefinite_length_array; /* For when ZCBOR_CANONICAL is defined */
 
-	memset(result, 0, result_len);
-
-	if (len == 0) {
-		*result_offs = additional;
+	if ((additional == ZCBOR_VALUE_IS_INDEFINITE_LENGTH) && (indefinite_length_array != NULL)) {
```

Comment:
What if its an indefinite length array and the caller passed NULL to `indefinite_length_array` ?